### ISSUE TITLE
AdGuard constantly asks to enter Administrator password after reboot

### ISSUE THREAD

**Issue by AlexandrPkhm** [AdguardTeam/AdguardForMac#1307]
### AdGuard version

2.12.0.1385 release

### OS version

macOS 13.6 Ventura and macOS 12.6.9 Monterey

### Support ticket ID

#815071

### Issue Details

Steps to reproduce:
1. Install AdGuard 
2. Complete the onboarding and enter Administrator's password for AdGuard to make changes
3. Reboot your Mac

### Expected Behavior

"AdGuard wants to make changes" prompt doesn't appear again

### Actual Behavior

"AdGuard wants to make changes" prompt appears after every reboot

### Screenshots

<details><summary>Screenshot 1:</summary>

![image](https://github.com/AdguardTeam/AdguardForMac/assets/109953785/503fcdb6-3c3c-4321-9cb5-53aaf1b1e34f)

</details>

### Possible solutions

Keep the setting `AdGuard Software Limited` located in `System settings → General → Login items → Allow in the Background` enabled.

**Comment by kikker902** [comment]
Needed to disable and enable AdGuard Software Limited located in System settings → General → Login items → Allow in the Background to solve this dialog on reboot. All now working as expected.

**Comment by kikker902** [comment]
Was too quick with the saying all works as expected now. A other reboot and AdGuard is asking for a password again. Perhaps something is not signed correctly in the code? Hope it will be fixed.

**Comment by AleksandrKalininAdGuard** [comment]
+ CRM 816530. Switching "Allow in the Background" toggle wasn't useful as well.
[adguard_logs_20230928074344.zip](https://github.com/AdguardTeam/AdguardForMac/files/12748602/adguard_logs_20230928074344.zip)
[serviceDump (1).txt](https://github.com/AdguardTeam/AdguardForMac/files/12748603/serviceDump.1.txt)
<img width="477" alt="Screenshot 2023-09-28 at 12 14 07 PM" src="https://github.com/AdguardTeam/AdguardForMac/assets/126869179/a6889c4d-da8c-45f5-a532-a3a4f1068608">

**Comment by AleksandrKalininAdGuard** [comment]
CRM 815071

**Comment by ihurin** [comment]
+1 CRM 815586

**Comment by AleksandrKalininAdGuard** [comment]
CRM 816537 logs and dump file in the ticket.

**Comment by AleksandrKalininAdGuard** [comment]
CRM 817199 logs and dump file are also in the ticket.

**Comment by enakr** [comment]
CRM 817405 dump files in the ticket.

**Comment by bobodoll131** [comment]
Same issue on macOS Catalina 10.15.7

**Comment by AleksandrKalininAdGuard** [comment]
CRM 819157 logs attached.

**Comment by Aydinv13** [comment]
@kikker902 @bobodoll131 

We've released the new Nightly-build of the app. Feedback from it will provide us with information required to fix the issue with our ad blocker app. Hence, please do the following step-by-step:
 
1) [Uninstall](https://adguard.com/kb/adguard-for-mac/installation/#regular) the AdGuard app.
2) Download and install the latest [Nightly-build](https://agrd.io/mac_nightly).
3) Reboot your device.
4) Reproduce the issue.
5) Immediately after that, go to the Terminal of your Mac and execute this command log show --last 3m --style compact --info --debug > start.log. After that you should receive start.logfile, please send it to us.

In addition, we'll need to check the regular logs. Please follow these steps to record a log file:

- Click on theAdGuard icon in the menu bar → ⚙︎ → Advanced → Logging → Reset logs....
- Go to Advanced → Logging → Logging level → Debug.
- Reproduce the issue.
- Head back to Advanced → Logging and click on the Export logs and System info... button.
- Save the archive, send it to us, and specify the time the issue was reproduced.
- Set the logging level back to Default, otherwise, the app will be recording the logs, which will be taking up too much space later on.

**Comment by kikker902** [comment]
Can confirm that the Nightly-build does solve the issue. Interesting observation is that now all login items are neatly grouped under one item under "Allow in the Background".

<img width="476" alt="CleanShot 2023-10-05 at 08 24 24@2x" src="https://github.com/AdguardTeam/AdguardForMac/assets/7262498/3758e8c9-04c8-4c13-b9f3-67f431b96352">

**Comment by TheWojtek** [comment]
> Can confirm that the Nightly-build does solve the issue.

As my report was the original reason for this, I double-confirm the solution for both Sonoma and Monterey. Also, the Montery 12.6.9 (now being a 12.7) machine I use was just simply upgraded with the nightly (without uninstalling from the instructions as I did on Sonoma) and this has resolved the issue as well.

**Comment by PunkNugget** [comment]
Just uninstalled AdGuard using this link:

https://adguard.com/kb/adguard-for-mac/installation/#uninstall

Actually I used a wonderful little app called: AppDelete. This is BY FAR the BEST app to delete ALL files. In fact all other apps that have their own "uninstall" app FOR that app, they usually don't delete all files. Again, this app does it. Even with the latest macOS Sonoma 14.0.

Unfortunately, I reinstalled AdGuard and now it doesn't work AT ALL. No matter how many times I try to unpause it to launch it, it keeps shutting down and putting it on pause. Very frustrating. I've tried contacting AdGuard tech support, but of course they're useless. Their bot system emails you to use that same stupid uninstall link. We already know that's not going to solve anything.

So, any help that I can get from anyone over here, would be awesome. Thank you.
